test/api/ExchangeApi/decodeOrders.test.ts

```diff
@@ -0,0 +1,97 @@
+import { decodeAuctionElements } from 'api/exchange/utils/decodeAuctionElements'
+import BN from 'bn.js'
+
+function _transformBnToString(order: {
+  sellTokenBalance: BN
+  priceNumerator: BN
+  priceDenominator: BN
+  remainingAmount: BN
+}): {
+  sellTokenBalance: string
+  priceNumerator: string
+  priceDenominator: string
+  remainingAmount: string
+} {
+  return {
+    ...order,
+    sellTokenBalance: order.sellTokenBalance.toString(),
+    priceNumerator: order.priceNumerator.toString(),
+    priceDenominator: order.priceDenominator.toString(),
+    remainingAmount: order.remainingAmount.toString(),
+  }
+}
+
+describe('Decode orders', () => {
+  test('Test no orders', async () => {
+    // GIVEN: An empty encoded order
+    const encodedOrders = '0x'
+
+    // WHEN: decoding
+    const orders = decodeAuctionElements(encodedOrders)
+
+    // THEN: The decoded orders are an empty array
+    expect(orders).toEqual([])
+  })
+
+  test('Test single order', async () => {
+    // GIVEN: A single order
+    const encodedOrders =
+      '0x0000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000040007005040240050402a0000000000000000000000000016e36000000000000000000de0b6b3a764000000000000000000000de0b6b3a7640000'
```

Comment:
Could you maybe split this into multiple lines?

Even though encoded, would help.

Reply:
I've made an encode function in the test and then I match this with the decoded result:

https://github.com/gnosis/dex-react/pull/371/commits/3e2bfc5973161f5d1ac3a2c1312ef4dece860e4b